cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/stats/BrokerCapacityStats.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.response.stats;
+
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Get broker level stats related to capacity in human readable format.
+ */
+public class BrokerCapacityStats extends BrokerStats {
+  private final List<SingleBrokerCapacityStats> _brokerCapacityStats;
+
+  public BrokerCapacityStats(KafkaCruiseControlConfig config) {
+    super(config);
+    _brokerCapacityStats = new ArrayList<>();
+  }
+
+  public void addSingleBrokerCapacityStats(String host, int id, boolean isEstimated, double diskCap, double cpuCap, double nwInCap,
+                                            double nwOutCap) {
+    SingleBrokerCapacityStats singleBrokerCapacityStats = 
+        new SingleBrokerCapacityStats(host, id, isEstimated, diskCap, cpuCap, nwInCap, nwOutCap);
+    _brokerCapacityStats.add(singleBrokerCapacityStats);
+    _isBrokerStatsEstimated = _isBrokerStatsEstimated || isEstimated;
+    _hostFieldLength = Math.max(_hostFieldLength, host.length());
+  }
+
+  @Override
+  protected void discardIrrelevantResponse() {
+    // Discard irrelevant response.
+    _brokerCapacityStats.clear();
+  }
+
+  /**
+   * Return an object that can be further be used to encode into JSON
+   */
+  @Override
+  public Map<String, Object> getJsonStructure() {
+    // broker level statistics
+    List<Map<String, Object>> brokerCapacityStats = new ArrayList<>(_brokerCapacityStats.size());
+    for (SingleBrokerCapacityStats stats : _brokerCapacityStats) {
```

Comment:
I think this can be rewritten with Java streams. You can transform each element with a map and then just a collector to collect them into a list. It seems that the original version also did this so I think it's fine to leave it as is though.

Reply:
Noted, that's definitely a better way but to keep it consistent with the existing code for now, I will leave it as is.